### DISCUSSION TITLE
Multiple chrome extensions?

### DISCUSSION THREAD

**Discussion by Vrackko** [seleniumbase/SeleniumBase#2672] [Q&A] [answered]
I have multiple extension paths. When i used one it was easy:

`
extension_path2 = f"extensions/fpd"
driver = Driver(uc=True, agent=user_agent, extension_dir=extension_path2, incognito=True, proxy=proxy_cred, uc_cdp_events=True)
`

But i have multiple extensions. Example:

`
extension_path2 = f"extensions/fpd"
extension_path3 = f"extensions/httphm"
`

I tried puting it in list but it doesnt work, it requiers string:
`
driver = Driver(uc=True, agent=user_agent, extension_dir=[extension_path2, extension_path3], incognito=True, proxy=proxy_cred, uc_cdp_events=True)
`

I also tried this solution:

`
extension_dirs = [extension_path2, extension_path3]
combined_extension_path = ','.join(extension_dirs)
driver = Driver(uc=True, agent=user_agent, extension_dir=combined_extension_path, incognito=True, proxy=proxy_cred, uc_cdp_events=True)
`
It renders the first path good, but second path tries to find it in my main google chrome path in extensions.

Any help with this?

**Comment by mdmintz** [comment, marked as answer]
Make sure you use a comma-separated string (no spaces) for loading multiple chrome extension directories in UC Mode.

https://github.com/seleniumbase/SeleniumBase/blob/332913648dd7ef60c60c364496cab5c5577fdcca/seleniumbase/plugins/driver_manager.py#L110

Also note that you can't use extensions with Incognito Mode or Guest Mode: https://github.com/seleniumbase/SeleniumBase/issues/2336#issuecomment-1837716219
(That restriction was made by Google: https://support.google.com/chrome_webstore/answer/2664769?hl=en)

**Reply by Vrackko** [reply]
Still got issue. Explanation of the issue below. Thanks for any help.

**Comment by Vrackko** [comment]
Thanks, but still got the error. It loads the link of first correctly but the link of the second doesnt. I get the error and it try to add aditional path to the my link.

`
driver = Driver(uc=True, agent=user_agent, extension_dir='extensions/fpd, extensions/httphm', proxy=proxy_cred, uc_cdp_events=True)
`

![image](https://github.com/seleniumbase/SeleniumBase/assets/145341123/85beec74-d31f-40f9-8960-0ab336c8891d)

**Reply by mdmintz** [reply]
I mentioned earlier in https://github.com/seleniumbase/SeleniumBase/discussions/2672#discussioncomment-9036333: "comma-separated string (**no spaces**)"
You had a space in your `extension_dir` string.
Also, `"Manifest file is missing or unreadable"` means that there may be something wrong with the formatting of your extension.

**Reply by Vrackko** [reply]
Thank you for your time to anwser. i still got same error pop up with or without the space. When i load them separately they are good, everything works, but every time i try to load them together the second one cant load. As you can see in the picture above it try to load extension from C:\Program Files\Google\Chrome\Application\123.0.6312.106\... which i dont want, path of my extension is:
C:\Users\HP\Desktop\No Com\extensions\httphm. Did you run into this bug before?

`
 extension_dir='extensions/httphm,extensions/fpd'
`

Edit:
I found a way but i dont think i will be able to export that:
`extension_dir='extensions/fpd,httphm'`

**extension/fpd** is in my script directory: C:\Users\HP\Desktop\No Com\extensions\fpd
**httphm** is in googles chrome directory:  C:\Program Files\Google\Chrome\Application\123.0.6312.107\httphm

So the script works but the path to the second plugin is bugged. I wanted to move this to the server but with this option it will be really  messy.

**Reply by zoennnnn** [reply]
Use `import os` and initialize the path of your extensions then concatenate your extensions with ',' in between in your `extension_dir=` argument.

`root = os.getcwd()`
`ext_10 = os.path.abspath(os.path.join(root, 'ext/ext1'))`
`ext_20 = os.path.abspath(os.path.join(root, 'ext/ext2'))`

`with SB(uc=True, extension_dir=f'{ext_10},{ext_20}') as sb:`

**Reply by shiraz-ahmed** [reply]
@Vrackko is right about that .when there are multiple extensions to be loaded only the first one works and there is error for the rest but when we try to load them individually it works without any problem .tried with the absolute path as well but still not working
